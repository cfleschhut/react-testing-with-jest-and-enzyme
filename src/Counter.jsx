/* eslint-disable react/sort-comp */

import React, { Component } from 'react';

class Counter extends Component {
  constructor(props) {
    super(props);

    this.state = { count: 0 };
  }

  makeIncrementer = amount => () =>
    this.setState(prevState => ({
      count: prevState.count + amount,
    }));

  increment = this.makeIncrementer(1);

  decrement = this.makeIncrementer(-1);

  render() {
    const { count } = this.state;

    return (
      <div>
        <p>
          Current count:
          <span>{count}</span>
        </p>
        <button type="button" onClick={this.increment} className="increment">
          +++
        </button>
        <button type="button" onClick={this.decrement} className="decrement">
          ---
        </button>

        <hr />
      </div>
    );
  }
}

export default Counter;
