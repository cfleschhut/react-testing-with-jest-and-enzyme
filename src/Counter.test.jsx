import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Counter from './Counter';

describe('Counter component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Counter />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('starts with a count of 0', () => {
    const wrapper = shallow(<Counter />);
    // const countState = wrapper.state().count;

    const text = wrapper.find('span').text();
    expect(text).toEqual('0');
  });

  it('can increment the count when the button is clicked', () => {
    const wrapper = shallow(<Counter />);
    const incrementBtn = wrapper.find('button.increment');

    incrementBtn.simulate('click');

    const text = wrapper.find('span').text();
    expect(text).toEqual('1');
  });

  it('can decrement the count when the button is clicked', () => {
    const wrapper = shallow(<Counter />);
    const decrementBtn = wrapper.find('button.decrement');

    decrementBtn.simulate('click');

    const text = wrapper.find('span').text();
    expect(text).toEqual('-1');
  });
});
