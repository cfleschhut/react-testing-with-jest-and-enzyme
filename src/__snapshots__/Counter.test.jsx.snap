// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component matches the snapshot 1`] = `
<div>
  <p>
    Current count:
    <span>
      0
    </span>
  </p>
  <button
    className="increment"
    onClick={[Function]}
    type="button"
  >
    +++
  </button>
  <button
    className="decrement"
    onClick={[Function]}
    type="button"
  >
    ---
  </button>
  <hr />
</div>
`;
